ovos_utils/file_utils.py

```
@@ -0,0 +1,240 @@
+    data_dir = os.path.expanduser(config.get('data_dir', "/opt/mycroft"))
+    filename = os.path.expanduser(os.path.join(data_dir, res_name))
+    if os.path.isfile(filename):
+        return filename
+
+    # look in ovos_utils package itself
+    found = resolve_ovos_resource_file(res_name)
+    if found:
+        return found
+
+    # Finally look for it in the source package
+    paths = [
+        "/opt/venvs/mycroft-core/lib/python3.7/site-packages/",  # mark1/2
+        "/opt/venvs/mycroft-core/lib/python3.4/site-packages/ ",
+        # old mark1 installs
+        "/home/pi/mycroft-core"  # picroft
+    ]
+    if root_path:
+        paths += [root_path]
+    for p in paths:
+        filename = os.path.join(p, 'mycroft', 'res', res_name)
+        filename = os.path.abspath(os.path.normpath(filename))
+        if os.path.isfile(filename):
+            return filename
+
+    return None  # Resource cannot be resolved
+
+
+def read_vocab_file(path):
+    """ Read voc file.
+
+        This reads a .voc file, stripping out empty lines comments and expand
+        parentheses. It returns each line as a list of all expanded
+        alternatives.
+
+        Args:
+            path (str): path to vocab file.
+
+        Returns:
+            List of Lists of strings.
+    """
+    vocab = []
+    with open(path, 'r', encoding='utf8') as voc_file:
+        for line in voc_file.readlines():
+            if line.startswith('#') or line.strip() == '':
+                continue
+            vocab.append(expand_options(line.lower()))
+    return vocab
+
+
+def load_regex_from_file(path, skill_id):
+    """Load regex from file
+    The regex is sent to the intent handler using the message bus
+
+    Args:
+        path:       path to vocabulary file (*.voc)
+        skill_id:   skill_id to the regex is tied to
+    """
+    from ovos_utils.intents.intent_service_interface import munge_regex
+
+    regexes = []
+    if path.endswith('.rx'):
```

else get mad? idk about returning `[]` on a bad file extension. that's a silent failure.

---

its not supposed to crash the whole thing, eg, it might just not exist for some lang, 

But again im just transplanting code from core... so fixing it is OOS, good to flag this stuff however